libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesControllerTest.kt

```diff
@@ -169,6 +169,100 @@ class RouteAlternativesControllerTest {
         verify(exactly = 1) { directionsSession.requestRoutes(any(), any()) }
     }
 
+    @Test
+    fun `trigger alternative route request based on user event`() = coroutineRule.runBlockingTest {
+        every { tripSession.getState() } returns TripSessionState.STARTED
+        mockRouteOptionsProvider(routeOptionsResultSuccess)
+        every { directionsSession.routes } returns listOf(
+            mockk {
+                every { routeIndex() } returns "0"
+                every { duration() } returns 1727.228
+            }
+        )
+        every { tripSession.getEnhancedLocation() } returns mockk {
+            every { latitude } returns -33.874308
+            every { longitude } returns 151.206087
+        }
+
+        val controller = mockController(
+            RouteAlternativesOptions.Builder()
+                .intervalMillis(TimeUnit.SECONDS.toMillis(10))
+                .build()
+        )
+
+        controller.register(routeAlternativesObserver)
+        controller.triggerAlternativeRequest()
+        controller.unregister(routeAlternativesObserver)
+
+        coroutineRule.testDispatcher.cleanupTestCoroutines()
+        verify(exactly = 1) { directionsSession.requestRoutes(any(), any()) }
+    }
+
+    @Test
+    fun `trigger alternative restarts timer and invoke api call`() = coroutineRule.runBlockingTest {
+        every { tripSession.getState() } returns TripSessionState.STARTED
+        mockRouteOptionsProvider(routeOptionsResultSuccess)
+        every { directionsSession.routes } returns listOf(
+            mockk {
+                every { routeIndex() } returns "0"
+                every { duration() } returns 1727.228
+            }
+        )
+        every { tripSession.getEnhancedLocation() } returns mockk {
+            every { latitude } returns -33.874308
+            every { longitude } returns 151.206087
+        }
+
+        val controller = mockController(
+            RouteAlternativesOptions.Builder()
+                .intervalMillis(TimeUnit.SECONDS.toMillis(10))
+                .build()
+        )
+
+        // Advance time to almost make a request, but cancel before. Prove that
+        // a new timer is started because there would be 2 calls otherwise.
+        controller.register(routeAlternativesObserver)
+        controller.triggerAlternativeRequest()
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.SECONDS.toMillis(10))
+        controller.unregister(routeAlternativesObserver)
```

Comment:
This test seems a little bit off. Wouldn't something like this test the behavior of restarting the timer better?
```suggestion
        controller.register(routeAlternativesObserver)
        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.SECONDS.toMillis(5))
        controller.triggerAlternativeRequest()
        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.SECONDS.toMillis(5))
        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.SECONDS.toMillis(10))
        controller.unregister(routeAlternativesObserver)
```
and assert that 2 requests are made and not 3. 3 would've been made if the timer wasn't restarted.